src/gc.c

```diff
@@ -1655,6 +1655,18 @@ void gc_free_pages(void)
     }
 }
 
+void gc_move_to_global_page_pool(jl_gc_page_stack_t *pgstack)
+{
+    while (1) {
+        jl_gc_pagemeta_t *pg = pop_lf_back(pgstack);
+        if (pg == NULL) {
+            break;
+        }
+        jl_gc_free_page(pg);
+        push_lf_back(&global_page_pool_freed, pg);
+    }
+}
+
```

Comment:
We likely need to subtract the memory we freed from `gc_heap_stats.heap_size`, as we do [here](https://github.com/JuliaLang/julia/blob/a6c656e6c47ff2b1237c92e90ba73ac267fc1dc0/src/gc.c#L1556C11-L1556C11).

Reply:
Didn't that linked line of code already do the subtraction for this though?

Reply:
No. `gc_sweep_page` doesn't subtract `heap_size` for nodes which are re-inserted into `page_metadata_buffered` of a given thread (i.e. pages still in use).

`gc_move_to_global_page_pool` is removing nodes from precisely `page_metadata_buffered` of one of the foreign threads that left.